lasagne/layers/special.py

```diff
@@ -207,46 +211,50 @@ class TransformerLayer(MergeLayer):
     ... nonlinearity=None)
     >>> l_trans = lasagne.layers.TransformerLayer(l_in, l_loc)
     """
-    def __init__(
-            self, input, localization_network, downsample_factor=1, **kwargs):
+    def __init__(self, incoming, localization_network, downsample_factor=1,
+                 **kwargs):
         super(TransformerLayer, self).__init__(
-            [input, localization_network], **kwargs)
-        self.downsample_factor = downsample_factor
+            [incoming, localization_network], **kwargs)
+        self.downsample_factor = as_tuple(downsample_factor, 2)
```

Comment:
"nothing" is not quite right, it will turn it into a tuple (which is important in case it's a generator) and check whether it's got the correct length. So it will do exactly the right thing :)